src/engine/strat_engine/serde_structs.rs

```diff
@@ -11,24 +11,20 @@
 // structs. These contain simple, serde-friendly data types, and we
 // can convert to or from them when saving our current state, or
 // restoring state from saved metadata.
-//
-// Look for "to_save" and "setup" methods, that either return or take
-// the below structs as parameters.
 
-use std::collections::HashMap;
-use std::path::PathBuf;
+use std::marker::Sized;
 
-use devicemapper::Sectors;
-
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct BlockDevSave {
-    pub devnode: PathBuf,
-    pub total_size: Sectors,
+/// Implements saving and restoring from metadata.
+pub trait DSerializable<T> {
+    fn to_save(&self) -> T;
+    fn setup(T) -> Self
+        where Self: Sized
+    {
+        unimplemented!()
+    }
```

Comment:
1. It makes it clear in the code what structs are saveable and what are not. This is better than grepping for ```to_save``` and ```setup``` methods as the header comment requires. It also clearly enforces what the signatures of these methods should be.

2. Do you have a suggestion?